time/plain_time.mbt

```diff
@@ -0,0 +1,337 @@
+  }
+  let mins = self.hour.to_int64() * minutes_per_hour + self.minute.to_int64()
+  let new_mins = (minutes % minutes_per_day + mins + minutes_per_day) % minutes_per_day
+  if mins == new_mins {
+    return Ok(self)
+  }
+  let new_hour = new_mins / minutes_per_hour
+  let new_minute = new_mins % minutes_per_hour
+  PlainTime::of(
+    new_hour.to_int(),
+    new_minute.to_int(),
+    self.second,
+    self.nanosecond,
+  )
+}
+
+/// Adds specified seconds to this time, and returns a new time.
+pub fn add_seconds(
+  self : PlainTime,
+  seconds : Int64
+) -> Result[PlainTime, String] {
+  if seconds == 0L {
+    return Ok(self)
+  }
+  let secs_of_day = self.second_of_day().to_int64()
+  let new_secs_of_day = (seconds % seconds_per_day + secs_of_day + seconds_per_day) %
+    seconds_per_day
+  if secs_of_day == new_secs_of_day {
+    return Ok(self)
+  }
+  let new_hour = new_secs_of_day / seconds_per_hour
+  let new_minute = new_secs_of_day / seconds_per_minute % minutes_per_hour
+  let new_second = new_secs_of_day % seconds_per_minute
+  PlainTime::of(
+    new_hour.to_int(),
+    new_minute.to_int(),
+    new_second.to_int(),
+    self.nanosecond,
+  )
+}
+
+/// Adds specified nanoseconds to this time, and returns a new time.
+pub fn add_nanoseconds(
+  self : PlainTime,
+  nanoseconds : Int64
+) -> Result[PlainTime, String] {
+  if nanoseconds == 0L {
+    return Ok(self)
+  }
+  let nanos = self.nanosecond_of_day()
+  let new_nanos = (nanoseconds % nanoseconds_per_day + nanos + nanoseconds_per_day) %
+    nanoseconds_per_day
+  if nanos == new_nanos {
+    return Ok(self)
+  }
+  PlainTime::from_nanosecond_of_day(new_nanos)
+}
+
+/// Adds a duration to this time, and returns a new time.
+pub fn add_duration(
+  self : PlainTime,
+  duration : Duration
+) -> Result[PlainTime, String] {
+  if duration.is_zero() {
+    return Ok(self)
+  }
+  self.add_nanoseconds(duration.to_nanoseconds())
+}
+
+/// Returns a new time with the specified hour.
+pub fn with_hour(self : PlainTime, hour : Int) -> Result[PlainTime, String] {
+  if hour == self.hour {
+    return Ok(self)
+  }
+  PlainTime::of(hour, self.minute, self.second, self.nanosecond)
+}
+
+/// Returns a new time with the specified minute.
+pub fn with_minute(self : PlainTime, minute : Int) -> Result[PlainTime, String] {
+  if minute == self.minute {
+    return Ok(self)
+  }
+  PlainTime::of(self.hour, minute, self.second, self.nanosecond)
+}
+
+/// Returns a new time with the specified second.
+pub fn with_second(self : PlainTime, second : Int) -> Result[PlainTime, String] {
+  if second == self.second {
+    return Ok(self)
+  }
+  PlainTime::of(self.hour, self.minute, second, self.nanosecond)
+}
+
+/// Returns a new time with the specified nanosecond.
+pub fn with_nanosecond(
+  self : PlainTime,
+  nanosecond : Int
+) -> Result[PlainTime, String] {
+  if nanosecond == self.nanosecond {
+    return Ok(self)
+  }
+  PlainTime::of(self.hour, self.minute, self.second, nanosecond)
```

Comment:
I think it would be beneficial for the compiler to handle branches more effectively in this way.
```suggestion
if nanosecond == self.nanosecond {
    return Ok(self)
} else {
    PlainTime::of(self.hour, self.minute, self.second, nanosecond)
}
```